packages/block-library/src/video/edit.native.js

```diff
@@ -235,7 +235,7 @@ class VideoEdit extends Component {
 			></MediaUpload>
 		);
 
-		if ( ! id ) {
+		if ( ! id && ! src ) {
```

Comment:
Did you explore and test the possibility of [updating this conditional match the web implementation](https://github.com/WordPress/gutenberg/issues/42515#issuecomment-1190935982)? I would think that would be the ideal outcome. 

From briefly testing that approach on an iPhone SE, I did not observe regressions  with the Video block. That said, it is likely worth testing it further if we pursue this approach. 

```diff
diff --git a/packages/block-library/src/video/edit.native.js b/packages/block-library/src/video/edit.native.js
index 434cd3c631..48b5692aee 100644
--- a/packages/block-library/src/video/edit.native.js
+++ b/packages/block-library/src/video/edit.native.js
@@ -172,7 +172,7 @@ class VideoEdit extends Component {
 				return;
 			}
 
-			setAttributes( { id: url, src: url } );
+			setAttributes( { src: url, id: undefined, poster: undefined } );
 		} else {
 			createErrorNotice( __( 'Invalid URL.' ) );
 		}
@@ -235,7 +235,7 @@ class VideoEdit extends Component {
 			></MediaUpload>
 		);
 
-		if ( ! id && ! src ) {
+		if ( ! src ) {
 			return (
 				<View style={ { flex: 1 } }>
 					<MediaPlaceholder
```

Reply:
@dcalhoun  Thanks for the feedback! After reviewing how `id` is used for other non-URL media options, I agree that `src` could and should be used here to match the web behavior. I have tested all video upload and embed block options for both platforms and did not note any regressions (including scenarios where videos originated from the web, e.g.: a draft video is uploaded via the web editor and then the draft is resumed and published on mobile). Once CI passes, I feel this would be ready for another review.